Windows CI Intermittent Fail in Setup Rust Toolchain

### Describe the bug

Over the last week, I have seen intermittent failures in Windows CI when trying to setup rust toolchain.

Here is a recent example:
https://github.com/apache/arrow-datafusion/actions/runs/8315114531/job/22753080370?pr=9623

And here is Windows CI working normally after just pushing an empty commit:
https://github.com/apache/arrow-datafusion/actions/runs/8315160379/job/22753186994?pr=9623

The error is 
```
 Error: Process completed with exit code 126.
```

### To Reproduce

Open a PR and run checks a few times.

### Expected behavior

Windows CI should pass/fail based on the test results not failing to get set up.

### Additional context

_No response_

--- a/.github/actions/setup-windows-builder/action.yaml
+++ b/.github/actions/setup-windows-builder/action.yaml
@@ -38,6 +38,7 @@ runs:
     - name: Setup Rust toolchain
       shell: bash
       run: |
+        # Avoid self update to avoid CI failures: https://github.com/apache/arrow-datafusion/issues/9653
         rustup toolchain install stable --no-self-update
         rustup default stable
         rustup component add rustfmt

--- a/.github/actions/setup-windows-builder/action.yaml
+++ b/.github/actions/setup-windows-builder/action.yaml
@@ -38,8 +38,7 @@ runs:
     - name: Setup Rust toolchain
       shell: bash
       run: |
-        rustup update stable
-        rustup toolchain install stable
+        rustup toolchain install stable --no-self-update
         rustup default stable
         rustup component add rustfmt
     - name: Configure rust runtime env

